### PR TITLE
107 I want to send and receive messages

### PR DESCRIPTION
**Pull request by cnstll**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [x] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: backend/app/src/channel/channel.gateway.ts [hunk 1]
```diff
@@ -74,37 +79,46 @@ export class ChannelGateway {
...
     @ConnectedSocket() clientSocket: Socket,
   ) {
-    const message: UserMessageDto = {
-      content: content,
-      senderId: senderId,
-    };
+    // console.log(messageInfo);
+    // const sockets = await this.server.fetchSockets();
+    // for (const s of sockets) {
+    //   console.log('id: ', s.id, 'rooms: ', s.rooms, 'data: ', s.data);
```
Comment (COTHSC) [inline backend/app/src/channel/channel.gateway.ts:105]: Do you still need this commented out bit?

File: backend/app/src/channel/channel.gateway.ts [hunk 2]
```diff
@@ -74,37 +79,46 @@ export class ChannelGateway {
...
+    // }
     const messageSaved = await this.channelService.storeMessage(
-      message,
-      channelId,
+      senderId,
+      messageInfo,
     );
-    messageSaved == null
```
Comment (COTHSC) [inline backend/app/src/channel/channel.gateway.ts:105]: I don't like ternaries.

**Review by COTHSC — APPROVED** [review]

File: frontend/app/src/components/section-components/chat/display-messages.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+              key={message.id}
+              content={message.content}
+              image={mockupImage}
+            />
+          ),
+        )}
+      {messageQuery.isSuccess &&
+        messageQuery.data &&
+        messageQuery.data.length === 0 && (
+          <div>No message in this channel yet! Don't be shy, send one!</div>
```
Comment (estelle-rcr) [inline frontend/app/src/components/section-components/chat/display-messages.tsx:101]: Quand on cree un channel ou quand on recharge la page sur l'URL d'un channel, le "No message in this...." ne s'affiche pas alors qu'il n'y a pas de message dans ces channels

Comment (estelle-rcr) [reply frontend/app/src/components/section-components/chat/display-messages.tsx:101]: + je n'arrive pas a envoyer des messages quand je suis dans cet etat ou rien n'est affiche. Quand je change de channel, le message s'affiche lorsque je reviens et je peux a nouveau envoyer des messages

File: frontend/app/src/components/section-components/chat/display-messages.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  return (
+    <div className="p-5 flex flex-col gap-4">
+      {messageQuery.isSuccess &&
+        messageQuery.data &&
+        messageQuery.data.length > 0 &&
+        messageQuery.data.map((message) =>
+          message.senderId === userId ? (
+            <CurrentUserMessage
+              key={message.id}
+              content={message.content}
```
Comment (estelle-rcr) [inline frontend/app/src/components/section-components/chat/display-messages.tsx:87]: On a des petits soucis d'affichage front sur les mots tres long, il faudrait utiliser un break word
image

File: frontend/app/src/components/section-components/chat/display-messages.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        messageQuery.data &&
+        messageQuery.data.length === 0 && (
+          <div>No message in this channel yet! Don't be shy, send one!</div>
+        )}
+      {messageQuery.isLoading && <LoadingSpinner />}
+      {messageQuery.isError && <div>Woops could not your messages :( </div>}
+    </div>
+  );
+}
+
```
Comment (estelle-rcr) [inline frontend/app/src/components/section-components/chat/display-messages.tsx:108]: L'affichage de la fenetre de message ne reste pas ancree en bas d'ecran (sur les derniers messages envoyes), ca serait mieux en termes d'usabilite mais on devrait peut etre faire un nouveau ticket d'enhancement ?

File: frontend/app/src/components/section-components/chat/display-messages.tsx (hunk 3 above)
Comment (cnstll) [reply frontend/app/src/components/section-components/chat/display-messages.tsx:111]: Fixed

File: frontend/app/src/components/section-components/chat/display-messages.tsx (hunk 5 above)
Comment (cnstll) [reply frontend/app/src/components/section-components/chat/display-messages.tsx:118]: #181 opened

**Comment by estelle-rcr** [comment]
Probleme de cache de l'avatarImg a réglé:
Uncaught TypeError: Cannot read properties of undefined (reading 'avatarImg')
    at display-messages.tsx:100:1
    at Array.map (<anonymous>)
    at DisplayMessages (display-messages.tsx:79:1)
    at renderWithHooks (react-dom.development.js:16305:1)
    at updateFunctionComponent (react-dom.development.js:19588:1)
    at beginWork (react-dom.development.js:21601:1)
    at beginWork$1 (react-dom.development.js:27426:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
    at workLoopSync (react-dom.development.js:26466:1)
    at renderRootSync (react-dom.development.js:26434:1)

**Pull request merged**